src/HotChocolate/Spatial/src/Types/ISerializeable.cs

```diff
@@ -0,0 +1,863 @@
+        }
+
+        protected IList<List<Coordinate>> ParseCoordinateParts(
+            ObjectValueNode valueNode,
+            int fieldIndex,
+            int partCount)
+        {
+            if (fieldIndex == -1)
+            {
+                throw InvalidStructure_CoordinatesIsMissing(null!);
+            }
+
+            IValueNode coordinatesValue = valueNode.Fields[fieldIndex].Value;
+
+            if (!(_coordinatesField.Type.ParseLiteral(coordinatesValue) is List<List<Coordinate>>
+                coordinates))
+            {
+                throw InvalidStructure_CoordinatesCannotBeNull(null!);
+            }
+
+            if (coordinates.Count < partCount)
+            {
+                throw InvalidStructure_CoordinatesOfWrongFormat(null!);
+            }
+
+            return coordinates;
+        }
+
+        protected bool TryParseCrs(
+            ObjectValueNode valueNode,
+            int fieldIndex,
+            out int srid)
+        {
+            if (fieldIndex > 0)
+            {
+                IValueNode crsField = valueNode.Fields[fieldIndex].Value;
+
+                if (_crsField.Type.ParseLiteral(crsField) is int parsedSrid)
+                {
+                    srid = parsedSrid;
+                    return true;
+                }
+            }
+
+            srid = 0;
+            return false;
+        }
+
+        protected (GeoJsonGeometryType type, object coordinates, int? crs)
+            ParseFields(ObjectValueNode obj)
+        {
+            GeoJsonGeometryType? type = null;
+            object? coordinates = null;
+            int? crs = null;
+
+            for (var i = 0; i < obj.Fields.Count; i++)
+            {
+                var fieldName = obj.Fields[i].Name.Value;
+                IValueNode syntaxNode = obj.Fields[i].Value;
+                if (TypeFieldName.EqualsInvariantIgnoreCase(fieldName))
+                {
+                    type = GeoJsonTypeSerializer.Default.ParseLiteral(syntaxNode)
+                        as GeoJsonGeometryType?;
+                }
+                else if (CoordinatesFieldName.EqualsInvariantIgnoreCase(fieldName))
+                {
+                    coordinates = GeoJsonPositionSerializer.Default.ParseLiteral(syntaxNode);
+                }
+                else if (CrsFieldName.EqualsInvariantIgnoreCase(fieldName) &&
+                    syntaxNode is IntValueNode node &&
+                    !node.IsNull())
+                {
+                    crs = node.ToInt32();
+                }
+            }
+
+            if (type == null)
+            {
+                throw new Exception();
+            }
+
+            if (coordinates == null)
+            {
+                throw new Exception();
+            }
+
+            return (type.Value, coordinates, crs);
+        }
+    }
+
+    public class GeoJsonTypeSerializer : IGeoJsonSerializer
+    {
+        private static readonly IDictionary<string, GeoJsonGeometryType> _nameLookup =
```

Comment:
I think in this case we can use one dictionary. If you use Dictionary instead if IDictionary you will have:

- ContainsKey / ContainsValue
- TryGetKey / TryGetValue

The reason why we use two in the enum is that there is an IEnumType and we have two keys.